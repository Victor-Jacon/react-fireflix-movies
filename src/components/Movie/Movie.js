import { useState } from "react";
import styled from "styled-components";
import { ContainerMovie, MovieTitle, WatchButton } from "./Movie.styles";

// JEST RENDER 1 - Para testar o componente, sugiro usar o export default function. Quando eu faço usando o componente funcional comum, dá vários erros. É uma preferência pessoal.
export default function Movie({ movie }) {
  {
    /* Consumir API - Renderizar 2
    Desestruturei os dois valores de dentro de movie (só usarei eles)

    img:  Passo o link do poster na props src.
    MovieTitle:   Passo o Title (título do filme)
    
    WatchButton
      href:   Fiz uma concatenação para montar uma busca personalizada no youtube. (busquei o título do filme no youtube)
    */
  }
  const { Poster, Title } = movie;

  {
    /* estilizacao condicional 1 */
  }
  const [watched, setWatched] = useState(false);

  return (
    <ContainerMovie>
      <div>
        <img src={Poster} alt={Title} />
      </div>

      <MovieTitle>{Title}</MovieTitle>

      {/* estilizacao condicional 2 - 
      onClick:  Salvo se a pessoa clicou pra assistir ou não. 
      watched:  Crio uma props que passa os dados para o styled-components. 
        se watched for true: já assisti (botão vai ficar cinza)
        se watched for false: não assisti ainda (botão vai ficar vermelho)
      */}
      <WatchButton
        href={
          "https://www.youtube.com/results?search_query=" +
          `${Title}` +
          "+watch+online"
        }
        target="_blank"
        rel="noreferrer"
        onClick={() => setWatched(true)}
        watched={watched}
      >
        Watch now
      </WatchButton>
    </ContainerMovie>
  );
}
