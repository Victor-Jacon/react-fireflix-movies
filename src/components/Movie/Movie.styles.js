import styled from "styled-components";

export const ContainerMovie = styled.div`
  display: flex;
  flex-direction: column;

  img {
    height: 300px;
  }
`;

export const MovieTitle = styled.h1`
  display: flex;
  align-items: center;

  font-size: 16px;
  color: white;
  margin-top: 16px;
  height: 42px;
`;

export const WatchButton = styled.a`
  display: flex;
  text-align: center;
  justify-content: center;
  align-items: center;
  margin-top: 16px;

  /* estilizacao condicional 3
    Recebo a props watched que passei na etapa anterior.
    watched é true: eu assisti, então vai ficar cinza.
    caso contrário: não assisti, vai ficar vermelho.
  */
  background-color: ${({ watched }) => (watched ? `#ccc` : `#a90000`)};
  border-radius: 8px;
  width: 120px;
  color: white;
  height: 42px;
`;
