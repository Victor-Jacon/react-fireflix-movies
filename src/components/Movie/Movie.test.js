import renderer from "react-test-renderer";
import Movie from "./Movie";

it("renders correctly", () => {
  // JEST RENDER 2 - Passamos um dado mockado para a props movie.
  const movie = {
    Title: "Star Wars: Episode IV - A New Hope",
    Year: "1977",
    imdbID: "tt0076759",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg",
  };

  // JEST RENDER 3 - Cria o componente + serializa para JSON + comparamos o componente gerado com o snapshot. Eles precisam "bater (match)".
  const MovieComponent = renderer.create(<Movie movie={movie} />).toJSON();
  expect(MovieComponent).toMatchSnapshot();
});
