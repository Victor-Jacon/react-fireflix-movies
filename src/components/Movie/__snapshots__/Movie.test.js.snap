// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bczRLJ kjXzdU"
>
  <div>
    <img
      alt="Star Wars: Episode IV - A New Hope"
      src="https://m.media-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"
    />
  </div>
  <h1
    className="sc-gsnTZi jFUJSo"
  >
    Star Wars: Episode IV - A New Hope
  </h1>
  <a
    className="sc-dkzDqf gjBgfZ"
    href="https://www.youtube.com/results?search_query=Star Wars: Episode IV - A New Hope+watch+online"
    onClick={[Function]}
    rel="noreferrer"
    target="_blank"
  >
    Watch now
  </a>
</div>
`;
